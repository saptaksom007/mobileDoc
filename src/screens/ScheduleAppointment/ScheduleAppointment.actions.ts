

export const ActionTypes = {
  SCHEDULE_APPOINTMENT: 'SCHEDULE_APPOINTMENT'
}

export default class Actions {
  static scheduleAppointment(scheduleAppointment: number) {
    return {
      type: ActionTypes.SCHEDULE_APPOINTMENT,
      payload: { scheduleAppointment }
    }
  }
}
