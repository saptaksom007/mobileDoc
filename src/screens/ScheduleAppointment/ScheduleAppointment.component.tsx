import React, { Component } from 'react'
import { View, ScrollView, InteractionManager } from 'react-native'
import { connect } from 'react-redux'
import { compose } from 'recompose'
import i18n from 'ex-react-native-i18n'
import { Card, ListItem } from 'react-native-elements'

import { findConversationForHealthSubjectUuid as findConversation } from 'common-docdok/lib/domain/messaging/selectors/findConversation'

import { momentWithLocales } from 'utilities/date'
import {
  Paragraph,
  Medium,
  Bold,
} from 'components/CustomText/CustomText.component'
import { fontFamily } from 'components/CustomText/CustomText.styles'
import Clinic from 'components/Clinic/Clinic.component'
import BookAppointmentLink from 'components/BookAppointmentLink/BookAppointmentLink.component'
import { withLoader } from 'screens/enhancers/withLoader'
import withSubtitle from 'screens/enhancers/withSubtitle'
import { Loader } from 'components/Loader/Loader.component'
import AppointmentConfirmationActions from 'screens/AppointmentConfirmation/AppointmentConfirmation.actions'
import Actions from 'screens/Appointment/Appointment.actions'

import styles from './ScheduleAppointment.styles'
import { Action } from 'redux'
import { ConversationDtoType } from 'common-docdok/lib/types'

interface Props {
  name: string
  clinic: string
  date: string
  street: string
  location: string
  phoneNumber: string
  dispatch(action: Action): void
  refreshing: boolean
  conversationId: number
  navigation: { state: { params: { appointmentId: number } } }
}

export class ScheduleAppointment extends Component<Props> {
  componentDidMount() {
    InteractionManager.runAfterInteractions(() => {
      const {
        navigation: {
          state: {
            params: { appointmentId },
          },
        },
        dispatch,
      } = this.props
      dispatch(Actions.appointmentRequested(appointmentId))
      dispatch(
        AppointmentConfirmationActions.markAsReadAppointmentConfirmation(
          appointmentId,
        ),
      )
    })
  }

  renderMessage = () => {
    const { date } = this.props
    return (
      <Medium style={styles.cardMessage}>
        <Bold>{i18n.t('scheduleAppointment.card.message')}</Bold>
        {'\n'}
        {date}
      </Medium>
    )
  }

  renderClinic = () => {
    const { clinic, phoneNumber, street, location } = this.props
    return (
      <Clinic
        name={clinic}
        contact={phoneNumber}
        street={street}
        location={location}
      />
    )
  }

  render() {
    const {
      clinic,
      name,
      refreshing,
      conversationId,
      navigation: {
        state: {
          params: { appointmentId },
        },
      },
    } = this.props

    if (refreshing || !name) {
      return <Loader />
    }

    return (
      <ScrollView contentContainerStyle={styles.container}>
        <View style={styles.messageContainer}>
          <Paragraph style={styles.scheduleMessage}>
            <Paragraph>
              {i18n.t('scheduleAppointment.salutation', {
                name,
                default: 'Hello,',
              })}
            </Paragraph>
            {'\n'}
            {'\n'}
            <Paragraph>
              {i18n.t('scheduleAppointment.message', {
                clinic,
              })}
            </Paragraph>
          </Paragraph>
        </View>
        <Card
          containerStyle={styles.cardContainerStyle}
          title={i18n.t('scheduleAppointment.card.title')}
          titleStyle={styles.cardTitle}
          fontFamily={fontFamily}
        >
          <ListItem
            title={this.renderMessage()}
            hideChevron
            containerStyle={styles.cardLastItem}
          />

          <ListItem
            title={this.renderClinic()}
            hideChevron
            containerStyle={[styles.cardLastItem, styles.cardLastItemNoBorder]}
          />
        </Card>
        <View style={styles.bottomLinks}>
          <BookAppointmentLink
            forceConversationId={conversationId}
            scheduleAppointment={appointmentId}
          />
        </View>
      </ScrollView>
    )
  }
}

function select({
  appointment: { data, refreshing },
  messaging: { conversations },
}: any) {
  const conv: ConversationDtoType | undefined =
    data &&
    Object.values(conversations as ConversationDtoType[]).find(
      findConversation(data.healthSubjectUuid),
    )
  return {
    name: (data && data.patientName) || '',
    clinic: (data && data.clinicName) || '',
    date:
      (data &&
        momentWithLocales(data.appointmentTime).format('dddd, MMMM Do')) ||
      '',
    street: (data && data.clinicStreet) || '',
    location: (data && data.clinicLocation) || '',
    phoneNumber: (data && data.clinicContact) || '',
    refreshing,
    conversationId: conv && conv.id,
  }
}

export default compose<Props, any>(
  connect(select),
  withLoader,
  withSubtitle('scheduleAppointment.subtitle'),
)(ScheduleAppointment)
