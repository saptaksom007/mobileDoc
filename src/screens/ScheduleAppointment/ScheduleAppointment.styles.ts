import { StyleSheet } from 'react-native'
import { Layout } from 'constants/Layout'
import { Color } from 'constants/Color'

export default StyleSheet.create({
  container: {
    height: Layout.window.height
  },
  messageContainer: {
    backgroundColor: '#fff',
    marginTop: 25,
    margin: 15,
    justifyContent: 'flex-start',
    alignItems: 'flex-start'
  },
  scheduleMessage: {
    textAlign: 'left'
  },
  cardContainerStyle: {
    marginTop: 10
  },
  cardTitle: {
    color: Color.lightText
  },
  cardMessage: {
    textAlign: 'center',
    width: 200,
    alignSelf: 'center'
  },
  cardLastItem: {
    paddingTop: 5,
    paddingBottom: 5
  },
  cardLastItemNoBorder: {
    borderBottomWidth: 0
  },
  bottomLinks: {
    padding: 20
  }
})
