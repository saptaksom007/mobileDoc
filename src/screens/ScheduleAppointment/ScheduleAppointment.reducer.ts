import { ActionTypes } from './ScheduleAppointment.actions'

const { SCHEDULE_APPOINTMENT } = ActionTypes

interface State {
  cacheAppointments: number[]
}

type Action = {
  type: keyof typeof ActionTypes
  payload: {
    scheduleAppointment: number
  }
}

const initialState: State = {
  cacheAppointments: [],
}

export default function scheduleappointment(
  state: State = initialState,
  action: Action,
) {
  switch (action.type) {
    case SCHEDULE_APPOINTMENT: {
      const cacheAppointments = state.cacheAppointments
      cacheAppointments.push(action.payload.scheduleAppointment)
      return {
        ...state,
        cacheAppointments: Array.from(new Set(cacheAppointments)),
      }
    }

    default:
      return state
  }
}
