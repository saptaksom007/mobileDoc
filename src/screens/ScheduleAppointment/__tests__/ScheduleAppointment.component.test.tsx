import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { ScheduleAppointment } from '../ScheduleAppointment.component'

jest.mock(
  'components/BookAppointmentLink/BookAppointmentLink.component',
  () => 'BookAppointmentLink',
)

it('<ScheduleAppointment/> render correctly', () => {
  const tree = renderer
    .create(
      <ScheduleAppointment
        name={'clinoxtest'}
        clinic={'clinoxtest'}
        date={''}
        street={'street'}
        location='location'
        phoneNumber='0303030303'
        conversationId={1}
        refreshing={false}
        dispatch={() => {}}
        navigation={{ state: { params: { appointmentId: 123 } } }}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
