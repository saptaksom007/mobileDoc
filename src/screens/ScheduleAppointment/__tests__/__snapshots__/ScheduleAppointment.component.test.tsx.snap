// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScheduleAppointment/> render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "height": 1334,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#fff",
          "justifyContent": "flex-start",
          "margin": 15,
          "marginTop": 25,
        }
      }
    >
      <Text
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 17.5,
              "lineHeight": 22,
              "overflow": "visible",
            },
            Object {
              "color": "rgba(27, 106, 158, 0.85)",
              "fontSize": 17.5,
              "lineHeight": 21,
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontSize": 17.5,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          scheduleAppointment.salutation
        </Text>
        

        

        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontSize": 17.5,
                "lineHeight": 21,
              },
              undefined,
            ]
          }
        >
          scheduleAppointment.message
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "padding": 20,
        }
      }
    >
      <BookAppointmentLink
        forceConversationId={1}
        scheduleAppointment={123}
      />
    </View>
  </View>
</RCTScrollView>
`;
